app/commands/scrape_command.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+class ScrapeCommand
+  attr_reader :keyword
+
+  def initialize(keyword_id:)
+    @keyword_id = keyword_id
+  end
+
+  # rubocop:disable Metrics/MethodLength
+  # rubocop:disable Metrics/AbcSize
+  def run
+    keyword = Keyword.find(@keyword_id)
+    html = Google::SearchService.search!(keyword.content)
+    result = Google::ParseService.new(html).call
+    persist_scrape_result keyword, result
+  rescue ActiveRecord::RecordNotFound => e
+    Rails.logger.error "[#{self.class.name}]: keyword doesn't exist
+    - keyword_id: #{@keyword_id}"
+
+    raise e
+  rescue GoogleScraperRuby::Errors::SearchServiceError => e
+    Rails.logger.error "[#{self.class.name}]: error while processing command
+    - keyword: #{keyword.content}
+    - error: #{e}"
+
+    raise e
+  end
+  # rubocop:enable Metrics/MethodLength
+  # rubocop:enable Metrics/AbcSize
+
+  private
+
+  def persist_scrape_result(keyword, parse_result)
```

Comment:
A more idionatic Ruby method name would be:

```suggestion
  def save_scrape_result(keyword, parse_result)
```

As it is consistent with the ActiveRecord API.

Reply:
Fixed in https://github.com/Goose97/google-scraper-ruby/pull/20/commits/defa7f47009b8c127be165fef468c28c5b83d6d9